code/game/turfs/open/lava.dm

```diff
@@ -137,9 +137,9 @@
 				O.resistance_flags |= FLAMMABLE //Even fireproof things burn up in lava
 			if(O.resistance_flags & FIRE_PROOF)
 				O.resistance_flags &= ~FIRE_PROOF
-			if(O.armor.fire > 50) //obj with 100% fire armor still get slowly burned away.
-				O.armor = O.armor.setRating(fire = 50)
-			O.fire_act(10000, 1000)
+			if(O.armor.fire > 50 * delta_time) //obj with 100% fire armor still get slowly burned away.
+				O.armor = O.armor.setRating(fire = 50 * delta_time)
```

Comment:
That doesn't make much sense, whatever the armor was it was cut down to 50 before ?

Reply:
You're right, that's pretty weird.